src/components/cc-addon-backups/cc-addon-backups.js

```diff
@@ -164,177 +192,256 @@ export class CcAddonBackups extends LitElement {
-              </cc-block-section>
-            </cc-block>
-          </div>
-        ` : ''}
       </cc-block>
     `;
   }
 
+  /**
+   * @param {AddonBackupsStateLoading} state
+   * @returns {TemplateResult}
+   * @private
+   */
+  _renderLoading (state) {
+    return html`
+      <div>
+        <span class="skeleton">
+          ${fakeString(150)}
+        </span>
+      </div>
+      ${this._renderBackups(state)}
+    `;
+  }
+
+  /**
+   * @param {AddonBackupsStateLoaded} state
+   * @returns {TemplateResult}
+   * @private
+   */
+  _renderLoaded (state) {
+    const hasData = state.backups.length > 0;
+
+    return html`
+      <div>${this._getDescription(state.providerId)}</div>
+
+      ${!hasData ? html`
+        <div class="cc-block_empty-msg">${i18n('cc-addon-backups.empty')}</div>
+      ` : ''}
+
+      ${hasData ? this._renderBackups(state) : ''}
+
+      ${this._overlayType === 'restore' ? this._renderRestoreOverlay(state.providerId, state.passwordForCommand) : ''}
+          
+      ${this._overlayType === 'delete' ? this._renderDeleteOverlay(state.providerId, state.passwordForCommand) : ''}
+    `;
+  }
+
+  /**
+   * @param {AddonBackupsStateLoaded | AddonBackupsStateLoading} state
+   * @private
+   */
+  _renderBackups (state) {
+    const skeleton = state.type === 'loading';
+    const areBtnsDisabled = state.type === 'loading' || this._overlayType != null;
+
+    const data = state.type === 'loaded' ? state : SKELETON_BACKUPS;
+    return html`
+      <div class="backup-list">
+          ${data.backups.map((backup) => html`
+            <div class="backup">
+              <span class="backup-icon"><cc-icon .icon=${iconBackup} size="lg"></cc-icon></span>
+              <span class="backup-text">
+                <span class="backup-text-details ${classMap({ skeleton })}">${this._getBackupText(backup)}</span>
+                <br>
+                ${ccLink(
+                  this._overlayType == null ? backup.url : null,
+                  this._getBackupLink(data.providerId),
+                  skeleton,
+                )}
+                <cc-button 
+                  link 
+                  ?disabled=${areBtnsDisabled} 
+                  ?skeleton=${skeleton} 
+                  @cc-button:click=${
+                    /** @param {CcButtonClickEvent} e */
+                    (e) => this._onOpenOverlay(e, 'restore', backup)
+                  }
+                >
+                  ${i18n('cc-addon-backups.restore.btn')}
+                </cc-button>
+                ${backup.deleteCommand != null ? html`
+                  <cc-button 
+                    link 
+                    ?disabled=${areBtnsDisabled} 
+                    ?skeleton=${skeleton} 
+                    @cc-button:click=${
+                      /** @param {CcButtonClickEvent} e */
+                      (e) => this._onOpenOverlay(e, 'delete', backup)
+                    }
```

Comment:
note: this kind of JSDoc is a bit verbose to my taste but maybe it's just me.

Reply:
@hsablonniere we agree with you but we have no alternative solution for this :shrug: